lib/Dialect/FIRRTL/FIRRTLFolds.cpp

```diff
@@ -67,6 +67,14 @@ static bool hasKnownWidthIntTypesAndNonZeroResult(Operation *op) {
   return true;
 }
 
+/// Return true if this value is 1 bit UInt.
+static bool valueIsOneBitUInt(Value op) {
+  auto t = op.getType().cast<UIntType>();
```

Comment:
Right, this is a bug, it should be:
`auto t = op.getType().dyn_cast<UIntType>();`

Otherwise it will crash on SInt's.  If this is only intended to be called on uint types, then please clarify the name of the function.